cmake/Modules/ConfigLibunwind.cmake

```
@@ -185,8 +185,10 @@ install(
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/timemory/libunwind/pkgconfig
     OPTIONAL)
 
+# Add include directories with BEFORE to ensure they come first in include search path
+# This ensures GNU libunwind headers are found before LLVM libunwind headers
 target_include_directories(
-    timemory-libunwind SYSTEM
+    timemory-libunwind BEFORE
```

```suggestion
    timemory-libunwind BEFORE SYSTEM
```

Maybe this is the best option. If `SYSTEM` flag is added compiler will add those directories with `-isystem` and not just `-I` when generating compile command, which will suppress any compile time warnings, such as unused variables ora any deprecated feature in **libunwind** code. If we build withoud SYSTEM flag and use `-Werror` option, any warning in **libunwind** will break our build. So this fix would take the both approaches: 
- Look these directories first within include path
- Treat them as system level headers

---

@marantic-amd  in this case valid options for INTERFACE targets is :

BEFORE - Controls search order priority
SYSTEM - Marks as system headers (suppresses warnings) 
BEFORE SYSTEM - Invalid 
SYSTEM BEFORE - Invalid 
Combining both keywords violates CMake's rule that INTERFACE targets can only have the SYSTEM keyword applied once in the entire target_include_directories command.

Generally a valid combinations are :
SYSTEM BEFORE - valid
SYSTEM AFTER - valid
BEFORE alone - valid
AFTER alone - valid
SYSTEM alone - valid
BEFORE SYSTEM - invalid (wrong order)
AFTER SYSTEM - invalid (wrong order)

---

If we use BEFORE SYSTEM then as per what I am reading, Compiler search order: regular includes (-I) then system includes (-isystem). Which means if ROCm's LLVM libunwind is in -I paths, we may hit the same issue. SYSTEM is mostly about warnings and may not reliably control shadowing if system-installed headers are in -I paths

We are already suppressing libunwind warnings [here](https://github.com/ROCm/timemory/blob/29037f3fdf86530578e2f58d4a542027054047bd/cmake/Modules/ConfigLibunwind.cmake#L122), so that should not be an issue.

So I think BEFORE alone is enough to fix the ordering of headers.

Should we also place a BEFORE [here](https://github.com/ROCm/timemory/blob/29037f3fdf86530578e2f58d4a542027054047bd/cmake/Modules/Packages.cmake#L1129) too?

---

No, we don't need to change this one.

The code you're looking at (lines 1126-1135 in Packages.cmake) is for the case when using a system-installed libunwind (TIMEMORY_USE_LIBUNWIND AND NOT TIMEMORY_BUILD_LIBUNWIND).

In this case:

It uses find_package(libunwind) to find the system library
The SYSTEM keyword alone is valid and just marks headers as system headers (suppresses compiler warnings)
There's no conflict with LLVM libunwind because you're explicitly using the found system libunwind
The problem we fixed was in ConfigLibunwind.cmake, which is used when building libunwind from source (TIMEMORY_BUILD_LIBUNWIND=ON). In that scenario, the built GNU libunwind headers needed to be searched before the system LLVM headers - hence we needed BEFORE.